workflows/theiacov/wf_theiacov_ont.wdl

```diff
@@ -150,73 +150,75 @@ workflow theiacov_ont {
             standardized_organism = organism_parameters.standardized_organism,
             seq_method = seq_method
         }
-      }
+      }      
       # consensus QC check
-      call consensus_qc_task.consensus_qc {
-        input:
-          assembly_fasta =  select_first([flu_track.irma_assembly_fasta, consensus.consensus_seq]),
-          reference_genome = organism_parameters.reference,
-          genome_length = organism_parameters.genome_length
-      }
-      # nanoplot for basic QC metrics
-      call nanoplot_task.nanoplot as nanoplot_raw {
-        input:
-          read1 = read1,
-          samplename = samplename,
-          est_genome_length = select_first([genome_length, consensus_qc.number_Total, organism_parameters.genome_length])
-      }
-      call nanoplot_task.nanoplot as nanoplot_clean {
-        input:
-          read1 = read_qc_trim.read1_clean,
-          samplename = samplename,
-          est_genome_length = select_first([genome_length, consensus_qc.number_Total, organism_parameters.genome_length])
-      }
-      # run organism-specific typing
-      if (organism_parameters.standardized_organism == "MPXV" || organism_parameters.standardized_organism == "sars-cov-2" || organism_parameters.standardized_organism == "rsv_a" || organism_parameters.standardized_organism == "rsv_b") { 
-        # tasks specific to either MPXV, sars-cov-2, rsv_a, or rsv_b
-        call nextclade_task.nextclade_v3 {
-          input:
-          genome_fasta = select_first([consensus.consensus_seq]),
-          dataset_name = organism_parameters.nextclade_dataset_name,
-          dataset_tag = organism_parameters.nextclade_dataset_tag
-        }
-        call nextclade_task.nextclade_output_parser {
+      if (defined(flu_track.irma_assembly_fasta) || defined(consensus.consensus_seq)) {
+        call consensus_qc_task.consensus_qc {
           input:
-          nextclade_tsv = nextclade_v3.nextclade_tsv,
-          organism = organism_parameters.standardized_organism
-        }
-      }    
-      if (organism_parameters.standardized_organism == "sars-cov-2") {
-        # sars-cov-2 specific tasks
-        call pangolin.pangolin4 {
+            assembly_fasta =  select_first([flu_track.irma_assembly_fasta, consensus.consensus_seq]),
+            reference_genome = organism_parameters.reference,
+            genome_length = organism_parameters.genome_length
+        }      
+        # nanoplot for basic QC metrics
+        call nanoplot_task.nanoplot as nanoplot_raw {
```

Comment:
It appears that the nanoplot call blocks are included within the conditional block starting on line 155, requiring some assembly FASTA file to be defined (either from IRMA or the iVar consensus FASTA) in order to run the tasks.

I think it would be beneficial if the nanoplot call blocks could be moved outside of this conditional, so that users still get the nanoplot outputs, even if an assembly is not generated.

Do you agree? If so could you please move the nanoplot call blocks above the conditional on 155 `if (defined(flu_track.irma_assembly_fasta) || defined(consensus.consensus_seq)) {` ?

Reply:
They are, but that's because assembly length (calculated from assembly) is included in the select first for nanoplot; willing to remove that from the select first but didn't want to make any changes without consideration

Reply:
Ah ok. Hmmm well for all organisms defined in the organism_parameters subwf (sarscov2, mpox, wnv, flu, rsva, rsvb, hiv) there will be a `genome_length` set automatically instead of relying upon the calculation from the assembly/consensus_qc task so it shouldn't fail at that step if no `consnsus_qc.number_Total` is defined.

Though I suppose for non-supported organisms if an assembly is not generated it would still fail. But I would hope that if a user is analyzing a non-supported organism, they would manually input the `genome_length` when running the workflow, preventing this potential failure.

Reply:
if this is outside the scope of this PR, we can make an issue and get to it later. Just wanted to raise this as an improvement

Reply:
I could make the switch, especially since QC values are the reason why we're doing this in the first place

Reply:
OK great. I think it would be beneficial to have QC values on the reads when assembly fails. 

I'm happy to test after you make those changes. I will test both Flu and sars-cov-2 samples through the wf. Please let me know when it's ready.